### DISCUSSION TITLE
Error when trying to run a function using local variable

### DISCUSSION THREAD

**Discussion by vitara1667** [ImageMagick/ImageMagick#5937] [Help]
### ImageMagick version

6.9.11-60

### Operating system, version and so on

Linuxmint Vanessa

### Description

I recently installed imagemagick on my PC for educational purposes. However when I tried to run a function which has local variables to accept variable inputs the system shows following error. Any help in solving the issue will be appreciated 
![IMG_0482](https://user-images.githubusercontent.com/121960333/210596026-3ac28247-67d8-401b-96c2-1dbe387113c0.jpeg)
![IMG_0483](https://user-images.githubusercontent.com/121960333/210596061-f069c72d-7509-477c-ad16-1f4fc986a650.jpeg)

**Comment by fmw42** [comment]
Please do not post images of code. We cannot easily copy your code to run it.  I am not familiar with the <( syntax.  Did you mean $(

You can write that command as one command line using parenthesis syntax. See https://imagemagick.org/Usage/basics/#parenthesis

**Comment by snibgo** [comment]
`$(...)` is command substitution. `<(...)` is *process* substitution.

Does the problem occur only with *local* variables, or with *any* variables?

I suspect that IM can't read images from process substitutions, because the process is asynchronous, so IM can't determine the size of the file until *after* it has been completely read, and it can't seek() into the file.

**Reply by vitara1667** [reply]
It occurs only for local variable
